### PR TITLE
adapt tests for provide-api-key annotation upstream

### PR DESCRIPTION
**Pull request by matthiaskubik**
- bumped gradle version due to SSL errors
- copied changes from https://github.com/apache/incubator-openwhisk/pull/4423

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -5,4 +5,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-bin.zip
```
Comment (rabbah) [inline gradle/wrapper/gradle-wrapper.properties]: Is this necessary? Might change required java version. I have to check but I think we use the same version in all the repos still.

Comment (rabbah) [reply gradle/wrapper/gradle-wrapper.properties]: Let me rephrase - if we’re doing it here, I’m assuming we will have to do it across all the repos.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -5,4 +5,4 @@ distributionBase=GRADLE_USER_HOME
...
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-bin.zip
```
Comment (rabbah) [inline gradle/wrapper/gradle-wrapper.properties]: With the change here do we have to make the same change across all the repos? I don’t know how many are on the old version. I’m assuming they will suffer the same ssl error, no?

Comment (matthiaskubik) [reply gradle/wrapper/gradle-wrapper.properties]: I'd be in favor of it but have not too much insight of what might break, so I rely on your advise...

**Comment by rabbah** [comment]
This PR should be merged _without_ squashing independent commits in case we need to roll back. Ideally orthogonal changes are in different PRs.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (rabbah) [reply gradle/wrapper/gradle-wrapper.properties]: https://github.com/apache/incubator-openwhisk/pull/4111/files#diff-fd9d61e37d939c941e7a62077f38d62dR6 the openwhisk repo is already on this version.

**Comment by rabbah** [comment]
@matthiaskubik I'm going to bump the gradle version to 4.10.2 in this PR https://github.com/apache/incubator-openwhisk-cli/pull/395. This PR then will address just the test changes. Sorry for the confusion. I'm trying to reduce the work for you.

**Comment by rabbah** [comment]
@matthiaskubik i rebased your PR to master to pick up the gradle bump and retitled your PR accordingly. Thanks for the contribution and fixing the tests.

**Comment by rabbah** [comment]
Thanks @matthiaskubik for the contribution.
